tests/unit/Imports/ProductImportTest.php

```diff
@@ -329,6 +346,178 @@ public function dataProviderTestGetStockProperties()
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => false,
+                    'value' => -5,
+                ],
+                'orderable_stock_value' => -5,
+                'backorder_enabled' => false,
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => true,
+                'quantity' => 0,
+                'backorders' => 'no',
+            ],
+        ];
+
+        $tests['simple, value positive, unlimited, backorder enabled'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => true,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => 5,
+                'backorder_enabled' => true,
+            ],
+            [
+                'in_stock' => true,
+                'manage_stock' => false,
+                'quantity' => null,
+                'backorders' => 'no', // backorder_enabled is ignored because it's unlimited so backorders are always no
+            ],
+        ];
+
+        $tests['simple, value positive, limited, backorder enabled'] = [
+            [
+                'type' => 'simple',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => false,
+                    'value' => 5,
+                ],
+                'orderable_stock_value' => 5,
+                'backorder_enabled' => true,
+            ],
+            [
+                'in_stock' => true, // always in stock because backorders are allowed
+                'manage_stock' => true,
+                'quantity' => 5,
+                'backorders' => 'yes',
+            ],
+        ];
+
+        $tests['configurable, value negative or 0, unlimited, backorder enabled'] = [
+            [
+                'type' => 'configurable',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => true,
+                    'value' => -5,
+                ],
+                'orderable_stock_value' => -5,
+                'backorder_enabled' => true,
+            ],
+            [
+                'in_stock' => false,
+                'manage_stock' => false,
+                'quantity' => null,
+                'backorders' => 'no', // backorder_enabled is ignored because it's unlimited so backorders are always no
+            ],
+        ];
+
+        $tests['configurable, value negative or 0, limited, backorder enabled'] = [
+            [
+                'type' => 'configurable',
+                'product_stock' => [
+                    'in_stock' => false,
+                    'unlimited' => false,
+                    'value' => -5,
+                ],
+                'orderable_stock_value' => -5,
```

Comment:
```suggestion
                'orderable_stock_value' => 0,
```
lets do 0 here because edge case